include/pybind11/detail/internals.h

```diff
@@ -148,18 +148,14 @@ struct override_hash {
 
 using instance_map = std::unordered_multimap<const void *, instance *>;
 
-// ignore: structure was padded due to alignment specifier
-PYBIND11_WARNING_PUSH
-PYBIND11_WARNING_DISABLE_MSVC(4324)
-
 // Instance map shards are used to reduce mutex contention in free-threaded Python.
-struct alignas(64) instance_map_shard {
+struct instance_map_shard {
     std::mutex mutex;
     instance_map registered_instances;
+    // alignas(64) would be better, but causes compile errors in macOS before 10.14 (see #5200)
+    char padding[64 - (sizeof(std::mutex) + sizeof(instance_map)) % 64];
```

Comment:
```suggestion
    PYBIND11_MAYBE_UNUSED char padding[64 - (sizeof(std::mutex) + sizeof(instance_map)) % 64];
```

Would this be a good idea? Compilers don't seem to care, though.

Reply:
I'm not sure it matters, but it seems fine to me?

Reply:
Will try to see if it is also fine. If not, will revert and go with it as is.

Thanks!